lib/p2p/types.ts

```diff
@@ -49,4 +49,15 @@ export type PoolConfig = {
    * It will be advertised with peers for them to try to connect to the server in the future.
    */
   addresses: string[];
+
+  /**
+   * An array of IP addresses or host names which can be used as a whitelist to limit connecting peers.
+   * Is will bypass all peers when it is empty.
+   */
+  whitelist: string[];
+
+  /**
+   * An array of IP addresses or host names which can be used as a blacklist to limit connecting peers.
+   */
+  blacklist: string[];
```

Comment:
Conceptually, I wonder whether we need a configurable blacklist of IP addresses within xud.  For one thing, I believe it would be much more efficient to block IPs before they even get to this point, banning them within the OS rather than within our app. When I wrote #458, I was thinking of banning IPs dynamically while xud is running if we get a lot of bad requests from them.

But in general, I don't any sort of fixed list will be all that effective since attackers can easily change IP addresses. My thinking is we should drop this config option altogether. Thoughts @kilrau ?